### PR TITLE
Renovate plugin

### PR DESCRIPTION
**Pull request by vim-zz**
Screenshot 2024-06-10 at 11 21 40

Screenshot 2024-06-10 at 14 45 29

### CODE REVIEW

**Review by MishaKav — DISMISSED** [review]

File: plugins/filters/compareSemver/index.js [hunk 1]
```diff
@@ -1,34 +1,30 @@
...
+ * @example {{ ["1.2.3", "1.2.1"] | compareSemver  == "patch" }}
  * @license MIT
 **/
 
 
-module.exports = (v1, v2, options = {}) => {
-  console.log("SEMVER", {v1, v2, options});
+module.exports = (v1, v2) => {
+  let options = {};
+  const { lexicographical = false, zeroExtend = true } = options;
```
[the remark this answers is not in the document]
Comment (vim-zz) [reply plugins/filters/compareSemver/index.js:21]: yes, exactly

[the remark this answers is not in the document]
Comment (MishaKav) [reply plugins/filters/compareSemver/index.js:21]: 1. 
```js
module.exports = (v1, v2, options = { lexicographical: false, zeroExtend: true }) => {
// just use
options.zeroExtend
```

2.
```js
module.exports = (v1, v2, options) => {
  const defaultOptions = { lexicographical: false, zeroExtend: true };
  options = { ...defaultOptions, ...options };

// just use
options.zeroExtend
```

[the remark this answers is not in the document]
Comment (vim-zz) [reply plugins/filters/compareSemver/index.js:21]: ok, i'll get rid of it

File: plugins/filters/extractRenovateVersionBump/extract_renovate_version_bump.cm [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - {{ bump == 'minor' }}
+      - {{ branch.name | includes(term="renovate") }}
+      - {{ branch.author | includes(term="renovate") }}
+    run:
+      - action: approve@v1
+      - action: add-comment@v1
+        args:
+          comment: |
+            Renovate `minor` version bumps are approved automatically.
+  bump_patch:
```
[the remark this answers is not in the document]
Comment (vim-zz) [reply plugins/filters/extractRenovateVersionBump/extract_renovate_version_bump.cm:16]: ok

File: plugins/filters/extractRenovateVersionBump/index.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        // remove trailing dot on to
+        if (to[to.length - 1] === ".") {
+          to = to.slice(0, -1);
+        }
+        results.push([to, from]);
+      }
+    } while(matches !== null);
+  }
+
+  return results;
```
[the remark this answers is not in the document]
Comment (vim-zz) [reply plugins/filters/extractRenovateVersionBump/index.js]: ok

File: plugins/filters/extractRenovateVersionBump/LICENSE [hunk 4]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2023 LinearB
```
Comment (BenLloydPearson) [inline plugins/filters/extractRenovateVersionBump/LICENSE:3]: Make sure you update this line to 2024

File: plugins/filters/compareMultiSemver/LICENSE [hunk 5]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2023 LinearB
```
Comment (BenLloydPearson) [inline plugins/filters/compareMultiSemver/LICENSE:3]: This should also be 2024

**Pull request merged**